src/lib/crypto/kimchi_pasta_snarky_backend/field.ml

```diff
@@ -228,7 +229,10 @@ module Make (F : Input_intf) :
         match j with
         | `String s ->
             let parsed_bigint =
-              if String.is_prefix ~prefix:"0x" s then Bigint.of_hex_string s
+              if
+                String.is_prefix ~prefix:"0x" s
+                || String.is_prefix ~prefix:"0X" s
```

Comment:
Didn't check, but it's important to make sure that the change is consistent with all other validators. I.e. that if  there's more than one place in the code that we parse hex of this particular format, we stay consistent between all these parsers.